gr-blocks/python/blocks/parse_file_metadata.py

```diff
@@ -75,12 +76,21 @@ def parse_header(p, VERBOSE=False):
         r = pmt.dict_ref(p, pmt.string_to_symbol("rx_time"), dump)
         secs = pmt.tuple_ref(r, 0)
         fracs = pmt.tuple_ref(r, 1)
-        secs = float(pmt.to_uint64(secs))
+        secs = pmt.to_uint64(secs)
         fracs = pmt.to_double(fracs)
-        t = secs + fracs
+        t = float(secs) + fracs
+        info["rx_time_secs"] = secs
+        info["rx_time_fracs"] = fracs
         info["rx_time"] = t
         if(VERBOSE):
-            print("Seconds: {0:.6f}".format(t))
+            with decimal.localcontext() as ctx:
+                # we need 36 decimal places in order not to lose precision
+                # since an uint64 can contain 20 decimal places and the
+                # resolution we should expect from the fracs duoble
+                # is the machine epsilon, which is ~1e-16
+                ctx.prec = 36
```

Comment:
26 would probably be enough here, since Unix timestamps won't have more than 10 digits until the year 2286. :smile:

36 will result in a lot of useless trailing digits:
```
>>> print(decimal.Decimal(1600000000) + decimal.Decimal(0.123456789012345))
1600000000.12345678901234499735828365
```

Reply:
The default precision is 28, which is enough, so you could also stick with that and limit the number of decimal places in output.

Reply:
You've convinced me that using 36 decimal places was an overkill, so I've changed this to use the default precision, and limited the output to 1e-16 resolution.